Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/opt/conda/lib/python3.9/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/conda/lib/python3.9/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/opt/conda/lib/python3.9/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/opt/conda/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/opt/conda/lib/python3.9/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/opt/conda/lib/python3.9/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/conda/lib/python3.9/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
coefficients = np.abs(model.coef_)
included = X_train.columns.values
indices = np.argsort(importances)[::-1]

fig_bar = px.bar(
    x=included[indices][0:5],
    y=importances[indices][0:5],
    title="Linear Coefficients in Ridge Regression",
    labels={"x": "Feature", "y": "Importance"},
)
fig_bar.show()
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn [35], line 3[0m
[1;32m      1[0m coefficients [38;5;241m=[39m np[38;5;241m.[39mabs(model[38;5;241m.[39mcoef_)
[1;32m      2[0m included [38;5;241m=[39m X_train[38;5;241m.[39mcolumns[38;5;241m.[39mvalues
[0;32m----> 3[0m indices [38;5;241m=[39m np[38;5;241m.[39margsort([43mimportances[49m)[::[38;5;241m-[39m[38;5;241m1[39m]
[1;32m      5[0m fig_bar [38;5;241m=[39m px[38;5;241m.[39mbar(
[1;32m      6[0m     x[38;5;241m=[39mincluded[indices][[38;5;241m0[39m:[38;5;241m5[39m],
[1;32m      7[0m     y[38;5;241m=[39mimportances[indices][[38;5;241m0[39m:[38;5;241m5[39m],
[1;32m      8[0m     title[38;5;241m=[39m[38;5;124m"[39m[38;5;124mLinear Coefficients in Ridge Regression[39m[38;5;124m"[39m,
[1;32m      9[0m     labels[38;5;241m=[39m{[38;5;124m"[39m[38;5;124mx[39m[38;5;124m"[39m: [38;5;124m"[39m[38;5;124mFeature[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124my[39m[38;5;124m"[39m: [38;5;124m"[39m[38;5;124mImportance[39m[38;5;124m"[39m},
[1;32m     10[0m )
[1;32m     11[0m fig_bar[38;5;241m.[39mshow()

[0;31mNameError[0m: name 'importances' is not defined
NameError: name 'importances' is not defined

